### DISCUSSION TITLE
Unstacking photos

### DISCUSSION THREAD

**Discussion by rickysarraf** [photoprism/photoprism#1592] [Q&A] [answered]
As a user, I'd like to unstack pictures in my collection. Upon full indexing, PhotoPrism has collated multiple pictures, which as per its criteria, are candidates for stacking. But the reality is, that they are not. This has been the common case, where I've digitized old vintage pictures of my parents. So, technically:

* They were taken at the same locaiton
* At the almost same time
* Through the same device

So lots of commonality from the stacking point of view, but they are very different pictures.

I went and disabled the stacking functionality in PhotoPrism, followed by a full re-index. But I still haven't been able to automatically `unstack` such pictures. Not sure if this is a bug of an expected behavior ? But it'd be nice to have PhotoPrism honor the settings, applied later, after the initial scan.

![image](https://user-images.githubusercontent.com/65480/135817987-bbdb198b-88cd-420e-853e-fb606321d629.png)

**Comment by lastzero** [comment, marked as answer]
Yes, it's expected as indexing doesn't break existing stacks. Unstacking is quite an expensive operation as new pictures need to be created in the index for each new file incl a complete extraction of metadata, creation of new thumbnails, conversion if it's a RAW and so on. The result may even depend on the order in which you unstack files if there are more than 2, especially where there are metadata sidecar files. Ideally, you make sure settings are good for your specific use case before you start indexing your library initially or reset your index and start again with a fresh database.

**Reply by rickysarraf** [reply]
Indeed. That's what is happening now. For every unstacked file, the full index (including face scanning) is run. But this is still better than resetting everything and starting again from scratch. But again, this can be a nightmare depending on how many photos got stacked. Like in my case, where I digitized printed photos, the stacks go in hundreds.

```
photoprism_1  | time="2021-10-04T16:11:14Z" level=info msg="File removed from stack"
photoprism_1  | time="2021-10-04T16:11:17Z" level=info msg="index: found 6 faces in IMG20170920184632.jpg [1.051482218s]"
photoprism_1  | time="2021-10-04T16:11:17Z" level=info msg="media: IMG20170920184632.jpg was taken at 2005-02-07 18:30:00 +0000 UTC (meta)"
photoprism_1  | time="2021-10-04T16:11:18Z" level=info msg="index: updated main jpg file “Pictures/Super Vintage/IMG20170920184632.jpg”"
photoprism_1  | time="2021-10-04T16:11:18Z" level=info msg="index: found 3 faces in IMG20170920180028.jpg [195.847397ms]"
photoprism_1  | time="2021-10-04T16:11:19Z" level=info msg="index: updated main jpg file “Pictures/Super Vintage/IMG20170920180028.jpg”"
photoprism_1  | time="2021-10-04T16:11:19Z" level=info msg="photos: found 1 result for id:pr0gnmr304an3v5x merged:true [1.49712ms]"
photoprism_1  | time="2021-10-04T16:11:19Z" level=info msg="photos: found 21 results for id:pr00p4b2kyrb1mnn merged:true [7.714431ms]"
```


So the *Stacking functionality* is an irreversible feature. Maybe this could be documented ?

**Reply by lastzero** [reply]
I think it is documented. You're welcome to help improving our docs 👍

**Reply by lastzero** [reply]
There's an existing feature request for an "Unstack All" button, it has just 4 likes and no sponsors though: https://github.com/photoprism/photoprism/issues/667

**Reply by rickysarraf** [reply]
I went ahead and reset the index, followed by a full rescan with `index -f -c` with the above settings as in the screenshot. But I still got a bunch of files stacked.

Upon manually unstacking them, following log was generated.

```
photoprism_1  | time="2021-10-05T08:53:35Z" level=info msg="index: found 2 faces in IMG_20190814_101259.a40b7301.jpg [1.260806399s]"
photoprism_1  | time="2021-10-05T08:53:35Z" level=info msg="media: IMG_20190814_101259.a40b7301.jpg was taken at 2019-08-05 10:36:20 +0000 UTC (meta)"
photoprism_1  | time="2021-10-05T08:53:35Z" level=info msg="classify: loading nasnet"
photoprism_1  | 2021-10-05 08:53:35.192454: I tensorflow/cc/saved_model/reader.cc:31] Reading SavedModel from: /photoprism/assets/nasnet
photoprism_1  | 2021-10-05 08:53:35.305048: I tensorflow/cc/saved_model/reader.cc:54] Reading meta graph with tags { photoprism }
photoprism_1  | 2021-10-05 08:53:35.981243: I tensorflow/cc/saved_model/loader.cc:202] Restoring SavedModel bundle.
photoprism_1  | 2021-10-05 08:53:38.101988: I tensorflow/cc/saved_model/loader.cc:311] SavedModel load for tags { photoprism }; Status: success. Took 2909535 microseconds.
photoprism_1  | time="2021-10-05T08:53:38Z" level=info msg="classify: loading labels from labels.txt"
photoprism_1  | time="2021-10-05T08:53:42Z" level=info msg="index: updated main jpg file Digikam_TO_BE_FILTERED/IMG_20190814_101259.a40b7301.jpg"
photoprism_1  | time="2021-10-05T08:53:43Z" level=info msg="index: found 2 faces in IMG20190805160620.jpg [144.347953ms]"
photoprism_1  | time="2021-10-05T08:53:43Z" level=info msg="index: updated main jpg file Digikam_TO_BE_FILTERED/IMG20190805160620.jpg"
photoprism_1  | time="2021-10-05T08:53:43Z" level=info msg="photos: found 1 result for id:pr0hy1912t9fcgx0 merged:true [1.350725ms]"
photoprism_1  | time="2021-10-05T08:53:43Z" level=info msg="photos: found 1 result for id:pqzunk7324fcuoy3 merged:true [1.46506ms]"
photoprism_1  | time="2021-10-05T08:53:43Z" level=info msg="File removed from stack"
```

**Reply by lastzero** [reply]
You disabled stacking and files still got stacked?

**Reply by rickysarraf** [reply]
Yes. Some of the files, not all. Of my colleciton of 15000 photos, I got like 50 such stacked files.

So, in the end, I wiped out the db, sidecar and album cache, and then did a fresh start, which is when it honored that setting. Now I don't have any stacked photos/files.

**Reply by lastzero** [reply]
Possible that files get stacked based on YAML backup files if files get the same photo assigned. Best to use our built-in reset command to start with a fresh index.

**Reply by lastzero** [reply]
Spent a lot of time this week to improve terminal command and config flag descriptions :)
